### PR TITLE
Add file size and hash code for Presto On Spark

### PR DESCRIPTION
**Pull request by viczhang861**
depended by https://github.com/facebookexternal/presto-facebook/pull/1470

see test plan in presto-facebook/pull/1470

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java [hunk 1]
```diff
@@ -293,7 +298,29 @@ public IPrestoSparkQueryExecution create(
         }
         else {
             checkArgument(sqlLocation.isPresent(), "sqlText or sqlLocation must be present");
-            sql = new String(metadataStorage.read(sqlLocation.get()), UTF_8);
+            byte[] sqlFileBytes = metadataStorage.read(sqlLocation.get());
+            if (sqlFileSizeInBytes.isPresent()) {
+                checkArgument(
```
Comment (arhimondr) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:303]: Let's classify these errors (as user error?)

**Review by arhimondr — APPROVED** [review]

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java (hunk 1 above)
Comment (viczhang861) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:303]: Actually user didn't set this argument,  the query file could be changed due to transport error, malicious attack etc. How about classify it as PrestoException but with error code in category SparkErrorCode?  Alerting may ignore user error category.

Comment (arhimondr) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:303]: Sounds good

**Review by arhimondr — APPROVED** [review]

**Comment by viczhang861** [comment]
@arhimondr This is ready to be merged, please also merge facebookexternal/presto-facebook#1470

**Pull request merged**